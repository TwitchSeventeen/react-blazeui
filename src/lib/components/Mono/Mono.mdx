---
name: Mono
---

import { Playground, PropsTable } from 'docz'
import Mono from './index';

# Mono

This styles text as monospaced. It can be used inline (ie it's not a block level element).

## Properties
<PropsTable of={Mono} />

## Basic usage
<Playground>
    <Mono>
        This class uses our mono font-family.
    </Mono>
</Playground>