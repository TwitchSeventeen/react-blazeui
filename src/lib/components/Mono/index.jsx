import React from 'react';
import PropTypes from 'prop-types';

function Mono (props) {
  const { children } = props;
  return (
      <span className="u-text--mono">
        { children }
      </span>
  );
}

Mono.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Mono.defaultProps = {
  children: null,
};

export default Mono;