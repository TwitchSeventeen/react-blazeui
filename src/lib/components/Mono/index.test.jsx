import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Mono from './index';

describe('Mono', () => {
  it('should be defined', () => {
    expect(Mono).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Mono>
          lorem ipsum dolor
        </Mono>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
