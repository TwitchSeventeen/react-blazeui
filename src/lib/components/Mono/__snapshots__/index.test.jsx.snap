// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mono renders correctly 1`] = `
<span
  className="u-text--mono"
>
  lorem ipsum dolor
</span>
`;
